Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -1123,5 +1130,79 @@ private void PrintNetworkInformation(NetworkConfiguration config, bool debug = t
                 _logger.Log(logLevel, "Filter list: {0}", _remoteAddressFilter.Select(s => s.Prefix + "/" + s.PrefixLength));
             }
         }
+
+        /// <summary>
+        /// Initializes the LAN IP address that the server advertises for cast receivers.
+        /// A LAN device like a Chromecast receiver needs an unique LAN address of the Jellyfin server.
+        /// For IPv4, a Private network address (like 192.168.0.0/16 )serves this purpose, and for IPv6
+        /// the Unique Local Address (ULA, fd00::/8) does. An IPv6 global address (2000::/3) also works
+        /// as unique server address but is ranked as less desirable than a local address.
+        /// See e.g. https://en.wikipedia.org/wiki/Private_network
+        ///
+        /// If multiple addresses are available:
+        /// - prefer Unique Local Addresses (ULA) or Site Local Addresses over other (global) addresses.
+        /// - all things equal, prefer IPv4 over IPv6 as the former in practice appears to be more stable.
+        /// </summary>
+        private void InitializeServerLanAddress(NetworkConfiguration config)
+        {
+            lock (_initLock)
+            {
+                int rank = 0;   // ranks most preferable ip address
+                IPAddress ipSrv = IPAddress.None;
+                NetworkInterface[] adapters = NetworkInterface.GetAllNetworkInterfaces();
+
+                foreach (var adapter in adapters)
+                {
+                    if (adapter.NetworkInterfaceType == NetworkInterfaceType.Loopback)
+                    {
+                        continue;
+                    }
+
+                    // check if the network interface serves as IPv4 gateway
+                    bool haveIP4GateWay = false;
+                    foreach (var address in adapter.GetIPProperties().GatewayAddresses)
+                    {
+                        haveIP4GateWay |= address.Address.AddressFamily == AddressFamily.InterNetwork;
+                    }
+
+                    // if the network interface is used as gateway, and has an acceptable IP address for this server,
+                    // then this address is assumed to be accessible by other devices on the LAN.
+                    foreach (var inf in _interfaces)
+                    {
+                        IPAddress ip = inf.Address;
+
+                        if (inf.Name.Equals(adapter.Name, StringComparison.OrdinalIgnoreCase) &&
```

Comment:
I would invert this `if`. The indentation gets pretty deep.

Reply:
Done!